src/main/java/cms/gov/madie/measure/repositories/MeasureSearchServiceImpl.java

```diff
@@ -210,36 +209,62 @@ public Page<MeasureListDTO> searchMeasuresByCriteria(
                 project(MeasureListDTO.class))
             .as("queryResults");
 
-    Aggregation pipeline = null;
+    Aggregation pipeline;
     if (appConfigService.isFlagEnabled(MadieFeatureFlag.MEASURE_SEARCH)) {
-      SortOperation sortOperation =
+      // Get all measureSetIds upfront based on the requested criteria
+      List<String> matchedMeasureSetIds =
+          mongoTemplate
+              .aggregate(
+                  newAggregation(
+                      lookupOperation, unwindOperation, matchOperation, group("measureSetId")),
+                  Measure.class,
+                  MeasureSetIdDTO.class)
+              .getMappedResults()
+              .stream()
+              .map(MeasureSetIdDTO::getId)
+              .collect(Collectors.toList());
+
+      MatchOperation matchMeasureSetIds =
+          match(Criteria.where("measureSetId").in(matchedMeasureSetIds));
```

Comment:
I think I understand the thought process here in trying to reduce the amount of documents being searched, but I think you'll end up seeing a performance drop due to the extra query and still get the same results.